Jonathan/Week_6/stations_challenge.py

```diff
@@ -0,0 +1,125 @@
+'''
+Using your SubwayStation class, instantiate a subway station with the info below. 
+Then run the show_info() method to make sure you get the station_name, location, and lines printed out
+
+station_name: '14th street'
+location: '14th street and 7th avenue'
+lines: ['1', '2', '3', 'L']
+'''
+
+station_union_square = SubwayStation('14th street', '14th street and 7th avenue', ['1', '2', '3', 'L'] )
+
+station_union_square.show_info()
+print()
+
+print('Question 3: Making the BusStation Class')
+
+'''
+Using the Station class below as the parent, make a child class called BusStation
+
+BusStation should:
+-have an additional attribute called 'routes' that is user-defined as a list during initialization. 
+    this will indicate where buses can go from this station. For example, ['DC', 'Philly', 'Pittsburgh']
+-have an additional attribute called 'open' that is always initalized as True (a boolean variable)
+-have additional methods called open_station() and close_station() to open and close the station
+-override the show_info() method from Station to display the bus routes and if the station is open, in addition to the station name and location
+'''
+class BusStation(Station):
+    def __init__(self, station_name, location, routes):
+        super().__init__(station_name, location)
+        self.routes = routes
+        self.open = True
+    
+    def open_station(self):
+        if self.open == True:
+            print(f'The {self.station_name} at {self.location} is open!')
+        else:
+            print(f'Sorry. The {self.station_name} at {self.location} is close.')
+
+    def show_info(self):
+        print(f'{self.station_name} station is located at {self.location}, and goes to the following routes: {self.routes}.')
+        self.open_station
+
+print()
+print('Question 4: Make an example bus station')
+'''
+Using your BusStation class, instantiate a bus station with the info below. 
+Then, run the show_info() method to make sure you get the station_name, location, routes, and whether the station is open printed out
+Then, demonstrate that you can close and open the bus station
+    i.e. that the show_info() method reflects correctly when the station is open versus closed
+
+station_name: 'NYC Megabus Stop'
+location: '34th street and 12th avenue'
+lines: ['Boston', 'DC', 'Philly']
+'''
+station_port_authority = BusStation('NYC Megabus Stop', '34th street and 12th avenue', ['Boston', 'DC', 'Philly'] )
+station_port_authority.show_info()
+
+
+print()
+print('Question 5: Importing your classes')
+
+'''
+Now, it's time to design a few more stations of your own in another script! 
+
+Make a new python script called "station_planning.py"
+    -In this script, *import* the classes you made in this challenge file
+    -Instantiate 3 more stations of your choosing (at least 1 bus and 1 subway)
+    -Feel free to make up names, locations, lines, and routes!
+'''
+
+from station_planning import Station, BusStation, SubwayStation
```

Comment:
This is great! But it needs to go in its own python file as described above, called station_planning.py